### ISSUE TITLE
Staging and Intermediate models for edx users

### ISSUE THREAD

**Issue by rachellougee** [mitodl/ol-data-platform#451]
# User Story
Create staging and intermediate models for edx users

# Description/Context
Source table - mitx_user_info_combo
  - user_id
  - id_map_hash_id
  - username
  - email
  - profile_*

The complete schema is available in 
https://github.com/MIT-IR/simeon/blob/main/simeon/upload/schemas/schema_user_info_combo.json

**Comment by rachellougee** [comment]
The user table we imported from BigQuery `raw__irx__edxorg__bigquery__user_info_combo` has 4 columns, and 

- [id_map_hash_id](https://github.com/MIT-IR/simeon/blob/main/simeon/upload/schemas/schema_user_info_combo.json#L196). - the same user can have multiple hash ID per system
- user_id
- username
- email

Here are some examples, use x for privacy
- Same user ID can have different hash IDs (this is a common case since users could have access to different platforms)

user_id | username | d_hash_map_id | Email
-- | -- | -- | --
30594148 | ph_xxxxxxx | aa3a1d2dc62b17a55bc54a | phxxxxx@gmail.com
30594148 | ph_xxxxxxx | 0c7f12b61382fb91f5e69fss | phxxxxx@gmail.com

-  Same user ID can have different email addresses and different hash IDs
    e.g this user has 4 entries in the table

user_id | username | d_hash_map_id | Email
-- | -- | -- | --
41245380 | asldfhasxxx | 0f6cf9a82877fe9e964a6 | jayxxxxxxx@gmail.com
41245380 | asldfhasxxx | e44cee4865a23f4295e6 | jayxxxxxxx@gmail.com
41245380 | asldfhasxxx | 06a1dfe311d9c32ffc481 | jayxxxxxxx@gmail.com
41245380 | asldfhasxxx | 0f6cf9a82877fe9e964a6| gedaxxxxx@87708b.com

- Same user ID has two entries, one with email and the other without email 

user_id | username | d_hash_map_id | Email
-- | -- | -- | --
4334679 | Vinotxxxx| dcde340a19545a08d05c14 | **vinothxxxxx@gmail.com**
4334679 | Vinotxxxx| dcde340a19545a08d05c14 |  

So the user ID is not unique in this table unlike the user's table from other sources like mitxonline, xpro

If we want to keep all 4 fields in this user table, the uniqueness would be **user_id + id_map_hash_id+user_email,** 

I think we should at least deduplicate some of the users like the last example? as I don't see why we need to keep a blank email for the same user ID with the same username and hash ID @dseaton what do you think?

**Comment by dseaton** [comment]
As a start, I am fine with deduplication.

I looked up user "Vinotxxxx". They have two enrollments in the "person_course" model. 

MITx/6.832x/3T2014 | 4334679 | Vinotxxxx
MITx/6.832x_2/3T2015 | 4334679 | Vinotxxxx

My guess is that somehow we failed to pull in the email address when parsing one of those courses and so the blank is being considered to make it a unique row. I'll keep looking into how IR is doing the aggregate process.

**Comment by rachellougee** [comment]
Besides `user_info_combo`, there are data issues in `user_info_name_country` as well
- Identical rows in `user_info_name_country`
  - same user_id, id_map_hash_id, profile_country, profile_mailing_address
- blank `user_id` 
   - this seems to be parsing errors or something like that. Values are stored as `b'xxxxx'`

To summarize the data issues that might need to be addressed from the source (IR repo?)
- Blank email in `user_info_combo` (last [example](https://github.com/mitodl/ol-data-platform/issues/451#issuecomment-1335524096) noted in above comment )
- Blank `user_id` in both `user_info_combo` and `user_info_name_country`
  - 67k rows that are blank 
  - seem to be parsing errors to me as noted above
  
I discussed these issues with @dseaton, documenting them here for reference

@pdpinch for FYI. If you need me to create a separate issue, please let me know.

**Comment by dseaton** [comment]
We are waiting on IRx to provide an updated user table. This will most likely be an aggregate of all the "[user_info_combo](https://github.com/MIT-IR/simeon/blob/main/simeon/upload/schemas/schema_user_info_combo.json)" tables available for each course.

**Comment by rachellougee** [comment]
once https://github.com/mitodl/ol-data-platform/issues/502  is done, I will stage the new table `mitx_user_info_combo`
